cli_tools/gce_vm_image_import/importer/disk_inspection_processor.go

```diff
@@ -0,0 +1,86 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package importer
+
+import (
+	"log"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/disk"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+type diskInspectionProcessor struct {
+	args          ImportArguments
+	diskInspector disk.Inspector
+}
+
+func (d *diskInspectionProcessor) process(pd persistentDisk) (persistentDisk, error) {
+	if !d.args.Inspect || d.diskInspector == nil {
+		return pd, nil
+	}
+
+	ir, err := d.inspectDisk(pd.uri)
+	if err != nil {
+		return pd, err
+	}
+
+	pd.isUEFICompatible = d.args.UefiCompatible || ir.HasEFIPartition
+	pd.isUEFIDetected = ir.HasEFIPartition
+	return pd, nil
+}
+
+func (d *diskInspectionProcessor) inspectDisk(uri string) (disk.InspectionResult, error) {
+	log.Printf("Running disk inspections on %v.", uri)
+	ir, err := d.diskInspector.Inspect(uri)
+	if err != nil {
+		log.Printf("Disk inspection error=%v", err)
+		return ir, daisy.Errf("Disk inspection error: %v", err)
```

Comment:
does this result in the error being printed twice?

Reply:
Yes, it's by design. It's similar to daisy workflow's behavior: print the error right away when it happens; then print again at the end of the cmd after cleanup is done.

Reply:
hmmmm.  curious. Well, it's probably better to stay consistent with the rest of the system. sgtm.